query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/schema/CreateTableWithOptions.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.querybuilder.schema;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.core.metadata.schema.ClusteringOrder;
+import com.datastax.oss.driver.api.querybuilder.BuildableQuery;
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+
+public interface CreateTableWithOptions
+    extends BuildableQuery, RelationStructure<CreateTableWithOptions> {
+
+  /** Enables COMPACT STORAGE in the CREATE TABLE statement. */
+  CreateTableWithOptions withCompactStorage();
+
+  /**
+   * Adds the provided CLUSTERING ORDER to the CREATE TABLE statement.
+   *
+   * <p>They will be appended in the iteration order of the provided map. If an ordering was already
+   * defined for a given identifier, it will be removed and the new ordering will appear in its
+   * position in the provided map.
+   */
+  CreateTableWithOptions withClusteringOrderByIds(Map<CqlIdentifier, ClusteringOrder> orderings);
```

Comment:
Why is this not in RelationStructure? As far as I can tell, CREATE MATERIALIZED VIEW can have a clustering order too.

Reply:
Yeah I think that makes sense to be moved there.  Will see if there's a reason why I didn't do that.

Reply:
Yeah that is the way to go. After this change `CreateTableWithOptions` will only have `withCompactStorage` and just primarily serve as a way to restrict the API to table options + compact storage.

Reply:
Ah, now I see a reason for this.  For `ALTER TABLE` these options shouldn't be available as you can't change clustering order posthoc.  I still think we want to retain these API methods for MVs, so we will need to break `RelationStructure` into two interfaces, (with one extending the other and providing clustering orders).